### ISSUE TITLE
Only share the timeportion of a calender (aka blocked view)

### ISSUE THREAD

**Issue by kaystrobach** [nextcloud/calendar#4782]
### Is your feature request related to a problem? Please describe.

As a user I want to share the times, where I have appointsments with someone else, but I do not want the thirdparty to know details like the location, notes, title, participants, etc..

### Describe the solution you'd like

Additional sharing option, which allows to mask all information, except that the calender has an event in a given time period.

### Describe alternatives you've considered

have a second calender which mirrors the events - is to much work - not sufficient

**Comment by ChristophWurst** [comment]
That is similar to https://github.com/nextcloud/calendar/issues/3185 but instead of a resource you want to view a user. I believe we had a ticket for that as well, yet I can't find it.

**Comment by kaystrobach** [comment]
I did not find one as well.

Thank you for pointing that out.

In #3185 they are explizitely mentioning a view as human viewable. I do see that more as a sharing option, which then can result in a human view.

On the sharing API only the provided Information has to be reduced to hide the sensitive data, displaying only the time slots.

**Comment by tcitworld** [comment]
If it's only the times you need to show you can create events in the shared calendar with the « when shared show only busy » setting.

Alas it's not possible to specify the visibility for the whole calendar. See #3992, #4086, #827, #725, #944, #531 (quite a list!) and https://github.com/nextcloud/server/issues/11214

**Comment by kaystrobach** [comment]
Need it for the whole calendar, not per event. But yea this can be an intermediate workaround

**Comment by sm8ps** [comment]
Judging from the many requests spread widely over different threads this seems to be a much sought-after feature as well as one that is not very likely to be added to Nextcloud very soon. 

Since I need the functionality myself, I have cobbled together a [script](https://github.com/sm8ps/shadowCalDAV) that makes a shadow copy of an ("internal") calendar to a "external" (shadow) calendar with veiled summary so that it can be published e.g. to the web.

This is no ideal solution by any means but it has provided useful so far. Feedback and suggestions welcome!

**Comment by kaystrobach** [comment]
This should be pretty easy. Wondering why no one picks this up.

**Comment by AlexanderZuehr** [comment]
We also need this feature because if we share calendars with customers they only should see the time and date (busy status) and not the rest of an appointment.
I know that if I create an appointment in the nextcloud calendar app I can choose what happens with this appointment if it is shared, but It is very time consuming to do so for every single one, furthermore if I manage the nextcloud calendar via caldav from my MacBook there is no option to choose this specific nextcloud share options, so every new appointment I add via caldav is completely visible to the shared person.

It would be the best if you can choose that the whole calendar and all its appointments are shown as busy if it is shared, so all problems would be fixed for me.

**Comment by kaystrobach** [comment]
You nailed it

**Comment by q-wertz** [comment]
I completely agree, that there should be a global setting for a share link but as far as I know the per appointment setting  is not a NC specific feature.
You also have it in Thunderbird and Evolution. So it's maybe a limitation of the Apple calendar... 
![image](https://github.com/nextcloud/calendar/assets/20948931/d9412b63-825a-46e9-bb16-7bb7d76a1b27)

**Comment by kaystrobach** [comment]
Any way the use case is there to have a busy non busy view, without the content of an appointment.

i would even go further:

* see and edit everything 
* See title and time only
* See busy only
* 
that should be a share setting.

**Comment by mdrobisch** [comment]
If someone is interested. I descibed a patch to get some of the behaviour mentioned here in a instance wise style. Please reference https://github.com/nextcloud/server/issues/11214#issuecomment-2213273846 

Together with an system- or calendarwise setting this might already the feature you want here.

**Comment by juengling** [comment]
> If it's only the times you need to show you can create events in the shared calendar with the « when shared show only busy » setting.

As I'm going to create some events by an own software, is there any chance to set this option via the caldav interface?

**Comment by kaystrobach** [comment]
Makes no sense, if you want someone to see the details and other to not see the details.

**Comment by juengling** [comment]
> Makes no sense, if you want someone to see the details and other to not see the details.

Right. But I'ld be the only one to see all details, so in this particular situation it would be sufficient. For a general purpose solution I agree to the above.
